chainer/training/extensions/snapshot_writers.py

```diff
@@ -57,6 +57,29 @@ def save(self, filename, outdir, target, savefun, **kwds):
             savefun(tmppath, target)
             shutil.move(tmppath, os.path.join(outdir, filename))
 
+        self.post_save(filename, outdir, target, savefun, **kwds)
+
+    def add_hook(self, hook_fun):
```

Comment:
The interface of the hook also needs more careful design.
Current form prohibits future extensions (like adding an argument to a hook function).
In order to avoid that, we should consider interface like `LinkHook` (it encupsulates arguments in a single instance).

Reply:
I see that idea. Possible solutions for this (adding cleanup to the writer) are:

1. Discuss the general interface for snapshot writers extension
2. Make this hook built-in and do not expose as an API, say like `def _add_cleanup_hook()`

I would prefer 2 as there are few possibility to extend the snapshot writers or we just haven't yet learned enough use cases for this. What do you think?

Reply:
Option 2. looks nice.
I assume users won't be able to insert their custom hooks, right?

Reply:
Done with 1f0c81e . Technically possible for users to add custom hooks, by you know, Python is hackable anywhere. I meant 'do not expose' as 1. undocumented, 2. method name starts with `_` .